### PR TITLE
core.py: tolerate bad response in watch_resources

### PR DESCRIPTION
**Pull request by mlangsdorf**
Since watch_resources operates in a loop, a single bad response
from the _get call shouldn't cancel operation.  Instead, catch
the exception from check_response() and increment a counter.
Only raise the exception to the next level if there are multiple
bad responses in a single watch_resources call.

Signed-off-by: Mark Langsdorf <mlangsdo@redhat.com>

### CODE REVIEW

File: osbs/core.py [hunk 1]
```diff
@@ -601,14 +602,31 @@ def delete_resource_quota(self, name):
...
         for retry in range(WATCH_RETRY):
-            response = self._get(url, stream=True, headers={'Connection': 'close'})
-            check_response(response)
+            try:
+                response = self._get(url, stream=True, headers={'Connection': 'close'})
+                check_response(response)
+            # we're already retrying, so there's no need to panic just because of a bad response
+            except OsbsResponseException as exc:
+                bad_responses += 1
+                if bad_responses > MAX_BAD_RESPONSES:
```
Comment (MartinBasti) [inline osbs/core.py]: I really like https://pypi.org/project/retry/ which does exactly the same.

Unfortunately it is not packaged in Fedora/Centos. But I really like the way with decorators.

I see there is `retry_on_exception` in `osbs-client/osbs/utils.py`. Could be used that decorator here?

Comment (mlangsdorf) [reply osbs/core.py]: retry_on_exception will retry the entire function.  That's overkill.  I just want to catch the exception and continue.

Comment (MartinBasti) [reply osbs/core.py]: ok

**Comment by MartinBasti** [comment]
No release notes, internal change only

**Pull request merged**